object-common.go

```diff
@@ -171,21 +172,28 @@ func cleanupDir(storage StorageAPI, volume, dirPath string) error {
 	// Function to delete entries recursively.
 	delFunc = func(entryPath string) error {
 		if !strings.HasSuffix(entryPath, slashSeparator) {
-			// No trailing "/" means that this is a file which can be deleted.
+			// Delete the file entry.
 			return storage.DeleteFile(volume, entryPath)
 		}
 		// If it's a directory, list and call delFunc() for each entry.
 		entries, err := storage.ListDir(volume, entryPath)
-		if err != nil {
-			if err == errFileNotFound {
-				// if dirPath prefix never existed.
-				return nil
-			}
+		if err != nil && err != errFileNotFound {
+			// If entryPath prefix never existed, safe to ignore.
+			// For any other errors fail.
 			return err
 		}
+		if len(entries) == 0 {
+			// Delete volume when entries are empty.
+			err = storage.DeleteVol(path.Join(volume, entryPath))
```

Comment:
This gives the impression that Volume and directories are same, Because, if there is an empty dir somewhere insde .minio/multipart upload it gets cleanedup by a DeleteVol call

Calling an explicit DeleteVol on `.minio/multipart` or `.minio/tmp` is fine because we treat them as volumes, but DeleteVol should not be done on anything inside it